src/plugins/kibana/public/settings/sections/indices/add_data_steps/pipeline_setup/lib/pipeline.js

```diff
@@ -0,0 +1,145 @@
+import _ from 'lodash';
+import * as ProcessorTypes from './processor_types';
+
+export default class Pipeline {
+
+  constructor() {
+    this.processors = [];
+    this.counter = 0;
+    this.input = {};
+    this.output = undefined;
+    this.dirty = false;
+
+    this._processorTypes = _.reduce(ProcessorTypes, (result, Type, key) => {
+      const instance = new Type();
+      result[instance.typeId] = Type;
+      return result;
+    }, {});
```

Comment:
You might consider making the processor class names and typeId match. As it stands each processor sort of has two different "types" which could be confusing. I know it's kind of stupid, but this is the state of javascript classes :/